src/types.ts

```diff
@@ -46,7 +46,7 @@ type TWallet = {
   ) => Promise<string /* | false */> // todo: sendTx reject => false
   disconnect: () => void
   estimateGas: (data: TransactionRequest) => Promise<BigNumber | undefined>
-  waitForTransaction: (transactionHash: string, confirmations?: number) => Promise<TransactionReceipt>
+  waitForTransaction: (transactionHash: string, confirmations?: number) => Promise<void>
```

Comment:
This is a change that fundamentally changes the logic of the function. We are often guided by the data we get from this function in router, so similar logic can be in the checkout. I would prefer to change the logic separately from restore Solana tx